arch/arm/src/kinetis/kinetis_edma.c

```diff
@@ -197,15 +199,16 @@ static struct kinetis_edmatcd_s *kinetis_tcd_alloc(void)
    * waiting.
    */
 
-  flags = enter_critical_section();
   nxsem_wait_uninterruptible(&g_edma.dsem);
 
+  flags = spin_lock_irqsave(&g_edma_lock);
+
```

Comment:
from nxsem_wait_uninterruptible to sq_remfirst must be in atomic region, we can not replace spinlock arbitrarily